runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -208,6 +209,55 @@ bool J9::TransformUtil::avoidFoldingInstanceField(
       "avoidFoldingInstanceField: symbol %p is never foldable (expected possibly foldable)\n",
       field);
 
+   if (fej9->isStable(field, cpIndex, owningMethod, comp) && !field->isFinal())
+      {
+      uintptr_t fieldAddress = object + fieldOffset;
+
+      TR::DataType loadType = field->getDataType();
+      switch (loadType)
+         {
+         case TR::Int32:
+            {
+            int32_t value = *(int32_t*)fieldAddress;
+            if (value == 0)
```

Comment:
While I realize that the "initial" value of a stable field is 0 and we do not want to fold the stable field unless it has been explicitly initialized, is'nt it possible that the "stable" value also happens to be 0 ? I am just asking because it seems that we won't fold in those cases since we cannot distinguish vs the initial case. 

Maybe this is not important due to some other property, i.e. that the stable values are expected to be non-zero ?

Reply:
Yes, the way I understand the definition of Stable, as long as a field has the default value, even if it was set explicitly in the user code or by constructor, it will not be folded.